applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h

```diff
@@ -200,15 +209,15 @@ class MapperVertexMorphing
             NodeVector neighbor_nodes( mMaxNumberOfNeighbors );
             std::vector<double> resulting_squared_distances( mMaxNumberOfNeighbors );
             unsigned int number_of_neighbors = mpSearchTree->SearchInRadius( node_i,
-                                                                             mFilterRadius, 
+                                                                             mFilterRadius,
                                                                              neighbor_nodes.begin(),
-                                                                             resulting_squared_distances.begin(), 
+                                                                             resulting_squared_distances.begin(),
                                                                              mMaxNumberOfNeighbors );
 
             std::vector<double> list_of_weights( number_of_neighbors, 0.0 );
             double sum_of_weights = 0.0;
 
-            ThrowWarningIfMaxNodeNeighborsReached( node_i, number_of_neighbors );                                                                               
+            ThrowWarningIfMaxNodeNeighborsReached( node_i, number_of_neighbors );
             ComputeWeightForAllNeighbors( node_i, neighbor_nodes, number_of_neighbors, list_of_weights, sum_of_weights );
```

Comment:
The answer highly depends to the type of the geometry. Bin used to be faster (about 5-40 X) for geometries that are cube shape. For a diagonal line which is the worse case it may become slower than KdTree. So least I would check it. Nevertheless we are completely rewriting the search. so I would suggest you to wait for new version before modifying it.